### PR TITLE
MueLu: interpolation order input changes for GeometricInterpolation

### PR DESCRIPTION
**Pull request by maxfirmbach**
@trilinos/muelu
@lucbv 
@mayrmt 

## Motivation
Right now, one has to specify the interpolation order of the structured algorithms inside muelu manually in the input file
for the StructuredAggregationFactory as well as in the GeometricInterpolationProlongation. With the changes, the
interpolation order value can be obtained directly from the StructuredAggregationFactory and thus makes the
usage of the GeometricInterpolationProlongation more intuitive.

## Testing
Changed the input files for the structured test cases in muelu, as well as the test cases in the corresponding research folder.
There's now a new variable called "structuredInterpolationOrder", which gets it's value from the StructuredAggregationFactory and specifically copies the "coarsening order".

### CODE REVIEW

**Review by lucbv — COMMENTED** [review]
It looks mostly good to me, the indentation in a few places is a bit off, it would be nice to have it fixed. Also I am guessing that the gold files will need to be rebased with the removal of this option in the prolongator factory?

File: packages/muelu/src/MueCentral/MueLu_FactoryManager_def.hpp [hunk 1]
```diff
@@ -194,14 +195,17 @@ namespace MueLu {
       if (varName == "Graph")                           return MUELU_KOKKOS_FACTORY(varName, CoalesceDropFactory, CoalesceDropFactory_kokkos);
       if (varName == "UnAmalgamationInfo")              return MUELU_KOKKOS_FACTORY(varName, AmalgamationFactory, AmalgamationFactory_kokkos);
       if (varName == "Aggregates")                      return MUELU_KOKKOS_FACTORY(varName, UncoupledAggregationFactory, UncoupledAggregationFactory_kokkos);
-      if (varName == "AggregateQualities")       return SetAndReturnDefaultFactory(varName, rcp(new AggregateQualityEstimateFactory()));
+      if (varName == "AggregateQualities")       		return SetAndReturnDefaultFactory(varName, rcp(new AggregateQualityEstimateFactory()));
```
Comment (mayrmt) [inline packages/muelu/src/MueCentral/MueLu_FactoryManager_def.hpp:198]: @maxfirmbach Please either revert this change in indentation or align it with the surrounding lines.

File: packages/muelu/src/Graph/StructuredAggregation/MueLu_StructuredAggregationFactory_kokkos_def.hpp [hunk 2]
```diff
@@ -248,10 +248,10 @@ namespace MueLu {
...
     }
 
-    Set(currentLevel, "lCoarseNodesPerDim",       geoData->getCoarseNodesPerDirArray());
-    Set(currentLevel, "indexManager",             geoData);
-    Set(currentLevel, "interpolationOrder",       interpolationOrder);
-    Set(currentLevel, "numDimensions",            numDimensions);
+    Set(currentLevel, "lCoarseNodesPerDim",       			geoData->getCoarseNodesPerDirArray());
+    Set(currentLevel, "indexManager",             			geoData);
+    Set(currentLevel, "structuredInterpolationOrder",       interpolationOrder);
+    Set(currentLevel, "numDimensions",            			numDimensions);
```
Comment (mayrmt) [inline packages/muelu/src/Graph/StructuredAggregation/MueLu_StructuredAggregationFactory_kokkos_def.hpp:254]: Please fix indentation.

File: packages/muelu/src/MueCentral/MueLu_FactoryManager_def.hpp [hunk 3]
```diff
@@ -194,14 +195,17 @@ namespace MueLu {
...
       if (varName == "CoarseMap")                       return MUELU_KOKKOS_FACTORY(varName, CoarseMapFactory, CoarseMapFactory_kokkos);
       if (varName == "DofsPerNode")                     return GetFactory("Graph");
       if (varName == "Filtering")                       return GetFactory("Graph");
       if (varName == "BlockNumber")                     return SetAndReturnDefaultFactory(varName, rcp(new InitialBlockNumberFactory()));
       if (varName == "LineDetection_VertLineIds")       return SetAndReturnDefaultFactory(varName, rcp(new LineDetectionFactory()));
       if (varName == "LineDetection_Layers")            return GetFactory("LineDetection_VertLineIds");
       if (varName == "CoarseNumZLayers")                return GetFactory("LineDetection_VertLineIds");
+      
+      // Structured
+      if (varName == "structuredInterpolationOrder")	return SetAndReturnDefaultFactory(varName, rcp(new StructuredAggregationFactory()));
```
Comment (mayrmt) [inline packages/muelu/src/MueCentral/MueLu_FactoryManager_def.hpp:208]:
```suggestion
      if (varName == "structuredInterpolationOrder")	  return SetAndReturnDefaultFactory(varName, rcp(new StructuredAggregationFactory()));
```

**Review by mayrmt — CHANGES_REQUESTED** [review]

**Comment by mayrmt** [comment]
@maxfirmbach Thanks for cleaning this up! Removing the necessity of duplicate input parameters will for sure help to avoid confusion in the future. 

As @lucbv already pointed out, please have another look at your changes. There are some places (and I have only marked a few of them), where indentation seems to be weird -- even in lines, whose content did not change.

**Review by mayrmt — DISMISSED** [review]

**Comment by mayrmt** [comment]
@maxfirmbach The autotester has come up with some errors. You can inspect the test results via the CDash dashboard by clicking on the link labelled "CDash Test Results for PR# 9153." in the [above autotester summary](https://github.com/trilinos/Trilinos/pull/9153#issuecomment-845168754). When you navigate to the failing test, you're lucky with this PR: the error message tells you pretty much what to do to fix the test. If you have further questions on that, let me know and we can discuss it next week.

(Btw, I'll convert this PR to "draft" status, so that it can be tested without requiring approval every time.)

**Review by mayrmt — DISMISSED** [review]

**Review by cgcgcg — CHANGES_REQUESTED** [review]
When you change the masterList.xml file, please run update_params.sh and then run the tests locally first.

**Comment by cgcgcg** [comment]
Hm, not good: https://testing.sandia.gov/cdash/test/72318766
Seems like left-overs from this PR make other PRs fail.

**Comment by maxfirmbach** [comment]
I added the .tex files and MueLu_MasterList.cpp, which where modified by ./update_params.sh. The ./testMasterList.sh runs without errors locally and I'm also able to make and build the user guide without problems. Additionally, the ctests also run without errors.

**Review by mayrmt — DISMISSED** [review]

**Review by mayrmt — APPROVED** [review]

**Review by cgcgcg — APPROVED** [review]
LGTM, merge away!

**Pull request merged**